server.h

```diff
@@ -278,8 +297,50 @@ class CTFOServer {
   }
 
   void OnMidichloriansEvent(const LogEntry& entry) {
-    // TODO(mzhurovich): update answers here.
-    static_cast<void>(entry);
+    std::unique_ptr<MidichloriansEvent> event;
+    if (entry.m == "POST") {
+      try {
+        ParseJSON(entry.b, event);
+        UpdateStateOnEvent(event);
+      } catch (const bricks::ParseJSONException&) {
+        // TODO(mzhurovich): Error logging.
+        std::cerr << "Parse JSON failed." << std::endl;
+      }
+    }
+  }
+
+  void UpdateStateOnEvent(const std::unique_ptr<MidichloriansEvent>& event) {
+    try {
+      const iOSGenericEvent& ge = dynamic_cast<const iOSGenericEvent&>(*event.get());
```

Comment:
The best explanation for why do you not get it is that I just said something stupid :-(

Everything's good -- I just misread it as dereferencing a `dynamic_cast<>` to a pointer. Sorry, my bad.